src/mlpack/methods/softmax_regression/softmax_regression_impl.hpp

```diff
@@ -41,20 +41,32 @@ size_t SoftmaxRegression::Classify(const VecType& point) const
   return size_t(label(0));
 }
 
-template<typename OptimizerType>
+template<typename OptimizerType, typename... CallbackTypes>
+double SoftmaxRegression::Train(const arma::mat& data,
+                                const arma::Row<size_t>& labels,
+                                const size_t numClasses,
+                                CallbackTypes&&... callbacks)
```

Comment:
Did you test the interface I proposed above, so a user can specify the optimizer as well, not only the type?